### PR TITLE
Update import credentials command for user management

### PR DESCRIPTION
**Pull request by ivov**
```
./packages/cli/bin/n8n import:credentials --input=file.json
./packages/cli/bin/n8n import:credentials --separate --input=myDir/
```

### CODE REVIEW

File: packages/cli/commands/import/credentials.ts [hunk 1]
```diff
@@ -56,60 +60,104 @@ export class ImportCredentialsCommand extends Command {
...
+			scope: 'global',
+		});
+
+		const owner = await Db.collections.User!.findOne({ globalRole });
+
+		if (owner) return owner;
+
+		const user = new User();
+
+		await Db.collections.User!.save(
```
Comment (csuermann) [inline packages/cli/commands/import/credentials.ts:150]: @ivov, I would not expect a function called "getInstanceOwner()" to perform a save operation.

Comment (ivov) [reply packages/cli/commands/import/credentials.ts:150]: Cornelius, thank you for your review.

I find it reasonable that `getInstanceOwner` would itself take care of this unlikely case to ensure the call always succeeds. Looking at other commands, we do not follow a pattern of breaking commands down into atomic methods. Also, `getInstanceOwner` was merged like this in the reset command: https://github.com/n8n-io/n8n/pull/2698

Please let me know if you recommend the change anyway and I'll update it everywhere: here, in reset, and in import creds.

Comment (csuermann) [reply packages/cli/commands/import/credentials.ts:150]: Thanks for elaborating, @ivov. I leave it to @BHesseldieck and @krynble to make the call for a refactoring. :-)

File: packages/cli/commands/import/credentials.ts [hunk 2]
```diff
@@ -56,60 +60,104 @@ export class ImportCredentialsCommand extends Command {
...
+
+					const sharedCredential = new SharedCredentials();
+
+					await Db.collections.SharedCredentials!.save(
+						Object.assign(sharedCredential, {
+							user: owner,
+							credentials: credential,
+							role: ownerCredentialRole,
+						}),
+					);
```
Comment (BHesseldieck) [inline packages/cli/commands/import/credentials.ts:100]: I know we didn't do it in commands yet but maybe we also should think about transactions?

If I am not mistaken we can create it like this:
```suggestion
					await Db.collections.SharedCredentials!.save({
							user: owner,
							credentials: credential,
							role: ownerCredentialRole,
						}),
					);
```

File: packages/cli/commands/import/credentials.ts [hunk 3]
```diff
@@ -56,60 +60,104 @@ export class ImportCredentialsCommand extends Command {
...
+
+					const sharedCredential = new SharedCredentials();
+
+					await Db.collections.SharedCredentials!.save(
+						Object.assign(sharedCredential, {
+							user: owner,
+							credentials: credentials[i],
+							role: ownerCredentialRole,
+						}),
+					);
```
Comment (BHesseldieck) [inline packages/cli/commands/import/credentials.ts:126]: maybe a transaction?

```suggestion
					const savedCredential = await Db.collections.Credentials!.save(credentials[i]);

					await Db.collections.SharedCredentials!.save( {
							user: owner,
							credentials: savedCredential,
							role: ownerCredentialRole,
						});
```

File: packages/cli/commands/import/credentials.ts [hunk 4]
```diff
@@ -56,60 +60,104 @@ export class ImportCredentialsCommand extends Command {
...
+
+		await Db.collections.User!.save(
+			Object.assign(user, {
+				firstName: 'default',
+				lastName: 'default',
+				email: null,
+				password: null,
+				resetPasswordToken: null,
+				...globalRole,
+			}),
```
Comment (BHesseldieck) [inline packages/cli/commands/import/credentials.ts:158]: if we do this we should just assign the role and let the database handle any defaults

**Review by BHesseldieck — CHANGES_REQUESTED** [review]

File: packages/cli/commands/import/credentials.ts [hunk 5]
```diff
@@ -56,60 +74,111 @@ export class ImportCredentialsCommand extends Command {
...
-						Credentials.prototype.setData.call(
-							fileContents[i],
-							fileContents[i].data,
-							encryptionKey,
-						);
+						Credentials.prototype.setData.call(credentials[i], credentials[i].data, encryptionKey);
 					}
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
-					await Db.collections.Credentials!.save(fileContents[i]);
+					await this.storeCredential(credentials[i], user);
```
Comment (BHesseldieck) [inline packages/cli/commands/import/credentials.ts:117]: let's wrap this part in a transaction and either set `this.transactionManager` or pass it as a param to `this.storeCredential` 👍🏼

File: packages/cli/commands/import/credentials.ts [hunk 6]
```diff
@@ -56,60 +74,111 @@ export class ImportCredentialsCommand extends Command {
...
+		}
+
+		return owner;
+	}
+
+	private async getAssignee(id: string) {
+		const user = await Db.collections.User!.findOne(id);
+
+		if (!user) {
+			throw new Error(`Failed to find user with ID ${id}. Are you sure this user exists?`);
```
Comment (BHesseldieck) [inline packages/cli/commands/import/credentials.ts:179]:
```suggestion
			throw new Error(`Failed to find user with ID ${id}`);
```

**Review by BHesseldieck — CHANGES_REQUESTED** [review]
awesome work!

File: packages/cli/commands/import/credentials.ts (hunk 6 above)
Comment (ivov) [reply packages/cli/commands/import/credentials.ts]: Then remove also the second sentence in this message?
`'File does not seem to contain workflows. Make sure the workflows are contained in an array.'`

Comment (ivov) [reply packages/cli/commands/import/credentials.ts]: Answered in mirror issue

File: packages/cli/commands/import/credentials.ts [hunk 7]
```diff
@@ -29,87 +45,161 @@ export class ImportCredentialsCommand extends Command {
 		separate: flags.boolean({
 			description: 'Imports *.json files from directory provided by --input',
 		}),
+		id: flags.string({
```
Comment (BHesseldieck) [inline packages/cli/commands/import/credentials.ts:48]: same as for other CLI command, would prefer the flag as `owner`, `user` or `ownerId/userId`

**Review by BHesseldieck — APPROVED** [review]
just the naming suggestion as for other CLI cmd

**Pull request merged**